src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -663,6 +663,109 @@ public int GetLength(MemoryPoolIterator end)
             }
         }
 
+        public static unsafe void FastByteCopy(byte[] srcArray, uint srcOffset, byte[] destArray, uint destOffset, uint byteCount)
+        {
+            Debug.Assert(srcOffset + byteCount <= srcArray.Length);
+            Debug.Assert(destOffset + byteCount <= destArray.Length);
+            
+            if (byteCount < 64)
+            {
+                // Buffer.BlockCopy is expensive for small copies.  Do it manually ourselves.
+                
+                fixed (byte* pSrc = srcArray)
+                fixed (byte* pDest = destArray)
```

Comment:
@geoffkizer Why is fixed() basically a no-op? I get that it's probably just marks the array and has no major impact on gc except in the relatively rare case when the collector would want to move the array in the middle of `FastByteCopy`.

I think the bigger issue is that the code generated for `fixed (byte* pDest = destArray)` will check if `destArray` is an null or zero length array while `fixed (byte* pDest = &destArray[0])` wont. At least according to #541 which is why we use this pattern in a few other places like `GetAsciiStringAsLong`.